source/shared/cpp/ObjectModel/UnknownElement.cpp

```diff
@@ -13,9 +13,29 @@ UnknownElement::UnknownElement() : BaseCardElement(CardElementType::Unknown)
 {
 }
 
+void UnknownElement::SetActualType(const std::string& actualType)
+{
+    m_actualType = actualType;
+}
+
+std::string UnknownElement::GetActualType() const
+{
+    return m_actualType;
+}
```

Comment:
What if BaseCardElements GetElementTypeString returned the actual type? I think that would be consistent with the way custom elements work.

Reply:
I was on the fence about this one. It felt a little weird to me to have it implicit. If you think it's weird to have it explicit, I can change it :)

---
In reply to: [251111068](https://github.com/Microsoft/AdaptiveCards/pull/2332#discussion_r251111068) [](ancestors = 251111068)